domain/topn_slow_query.go

```diff
@@ -115,6 +115,11 @@ type topNSlowQueries struct {
 	period   time.Duration
 	ch       chan *SlowQueryInfo
 	msgCh    chan *showSlowMessage
+
+	mu struct {
+		sync.RWMutex
+		closed bool
```

Comment:
how about use `atomic` instead of `mutex`.

Reply:
atomic can't protect multiple operations. @crazycs520 

```
modify atomic flag
close(ch)

check atomic flag
<-ch
```

Take this order for example:

1. check atomic flag = success
2. modify atomic flag = closed
3. close(ch)
4. <-ch  panic!